### PR TITLE
Make project REUSE compliant

### LINKED ISSUE #632: Make project REUSE compliant
See:
- https://github.com/VILLASframework/fpga/pull/66
- https://reuse.software/

### PR DESCRIPTION
**Pull request by stv0g**
Fixes #632

### CODE REVIEW

**Comment by stv0g** [comment]
@PJungkamp I've updated the PR with fixes for the missing trailing dots and wrong email addresses as you have requested :)

**Comment by PJungkamp** [comment]
The "Use C++ style comment" commit removes all the Doxygen-style `/**` and `///` comments and the Doxygen-style `@copyright` annotations and similar. We should probably also remove the other Doxygen-style `@` keywords (I've seen `@todo` and `@diafile` during my review). @stv0g Do you think this belongs to the scope of this PR?

**Comment by stv0g** [comment]
Hi @PJungkamp,

I agree. We should try to get rid of those as well. However, I would try to do this in the next PR as this one originally focused on the REUSE compliance only.

File: src/villas-zmq-keygen.cpp [hunk 1]
```diff
@@ -1,31 +1,9 @@
...
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
- * FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
- * License for more details.
- *
- * You should have received a copy of the GNU Lesser General Public License
- * along with this program.  If not, see <http://www.gnu.org/licenses/>.
- *********************************************************************************/
+ * Author: Steffen Vogel <post@steffenvogel.de>
+ * SPDX-FileCopyrightText: 2014-2023 Institute for Automation of Complex Power Systems, RWTH Aachen University
+ * SPDX-License-Identifier: Apache-2.0
```
Comment (PJungkamp) [inline src/villas-zmq-keygen.cpp:5]: Isn't this LGPL?

**Review by PJungkamp — CHANGES_REQUESTED** [review]
Finally done! Seems mostly fine. There are still some comments regarding the consistent annotations of email on contributors and I marked some more style errors when I came across those.

Could you take the annotations that I made as a hint and do a sanity check using code search to see whether there are similar style fixes I missed?

Please fix the license of `villas-zmq-keygen` and mark it with the `WITHOUT_GPL` CMake option while you're at it? This PR can then also resolve https://github.com/VILLASframework/node/issues/709.

File: clients/opal/models/send_receive/include/utils.h [hunk 2]
```diff
@@ -1,16 +1,15 @@
...
  *
- * @file
- * @author Steffen Vogel <post@steffenvogel.de>
- * @author Mathieu Dubé-Dallaire
- * @copyright 2003, OPAL-RT Technologies inc
- * @copyright 2014-2022, Institute for Automation of Complex Power Systems, EONERC
- * @license Apache 2.0
- *********************************************************************************/
+ * Author: Steffen Vogel <post@steffenvogel.de>
+ * Author: Mathieu Dubé-Dallaire
```
[the remark this answers is not in the document]
Comment (stv0g) [reply clients/opal/models/send_receive/include/utils.h:4]: I've checked. Hey is apparently not working for OPAL-RT anymore.
I got his name from the original example model. So I would say we can keep it as is.

File: clients/opal/models/send_receive/src/utils.c [hunk 3]
```diff
@@ -1,11 +1,11 @@
...
+/* Configure scheduler.
  *
- * @author Steffen Vogel <post@steffenvogel.de>
- * @author Mathieu Dubé-Dallaire
- * @copyright 2003, OPAL-RT Technologies inc
- * @copyright 2014-2022, Institute for Automation of Complex Power Systems, EONERC
- * @license Apache 2.0
- *********************************************************************************/
+ * Author: Steffen Vogel <post@steffenvogel.de>
+ * Author: Mathieu Dubé-Dallaire
```
[the remark this answers is not in the document]
Comment (stv0g) [reply clients/opal/models/send_receive/src/utils.c:4]: Same as in https://github.com/VILLASframework/node/pull/708/#discussion_r1315040763, we unfortunately dont have an email address.

File: etc/loopback.conf
[the remark this answers is not in the document]
Comment (stv0g) [reply etc/loopback.conf:1]: Fixed in fd679c0c

File: etc/websocket-demo.conf
[the remark this answers is not in the document]
Comment (stv0g) [reply etc/websocket-demo.conf:1]: Fixed in fd679c0c

File: src/villas-zmq-keygen.cpp (hunk 1 above)
Comment (stv0g) [reply src/villas-zmq-keygen.cpp:5]: No, I dont think so. There is no external code in this file. We just call a function from libzmq.

File: webrtc_a.conf
[the remark this answers is not in the document]
Comment (stv0g) [reply webrtc_a.conf:1]: Removed the file in 63514e01

File: doc/index.html [hunk 4]
```diff
@@ -1,3 +1,7 @@
+<!-- 
+SPDX-FileCopyrightText: 2014-2023 Institute for Automation of Complex Power Systems, RWTH Aachen University
+SPDX-License-Identifier: Apache-2.0
+ -->
 <!DOCTYPE html>
```
[the remark this answers is not in the document]
Comment (stv0g) [reply doc/index.html]: Fixed in 5755b222

File: doc/theme/layout.xml [hunk 5]
```diff
@@ -1,3 +1,7 @@
+<!--
```
[the remark this answers is not in the document]
Comment (stv0g) [reply doc/theme/layout.xml]: I have removed Doxygen now completely from the repo in d75efcb4

File: CHANGELOG.md
[the remark this answers is not in the document]
Comment (stv0g) [reply CHANGELOG.md:1]: Removed in 2fb0076d

File: tests/integration/README.md [hunk 6]
```diff
@@ -18,3 +18,8 @@ Example:
 ```
 $ BUILDDIR=/VILLASnode/build/ /VILLASnode/tools/integration-tests.sh -f pipe-loopback-socket -v
 ```
+
```
[the remark this answers is not in the document]
Comment (stv0g) [reply tests/integration/README.md:21]: Fixed in 68200a21

File: etc/labs/lab18.conf
[the remark this answers is not in the document]
Comment (stv0g) [reply etc/labs/lab18.conf:1]: Oh yeah. Probably yes. But I would like to avoid touching the history now..

**Comment by stv0g** [comment]
Okay, I think I am done :)

Please note that the commit hashes in my responses are broken as I had to do a rebase.
I guess GitHub might still have cached the commits, but the actual commit hashes haved changed due to a necessary rebase (failing DCO check)

**Comment by stv0g** [comment]
@windrad6 @n-eiling It would be nice if I can also get your approvals for this PR as you are codeowners of some of the files which I have touched.

File: include/villas/kernel/if.hpp [hunk 7]
```diff
@@ -48,7 +48,7 @@ class Interface {
 	int affinity;				// IRQ / Core Affinity of this interface.
 
 	std::list<int> irqs;			// List of IRQs of the NIC.
-	std::list<node::Node *> nodes;	// List of nodes which use this interface.
+	std::list<node::Node *> nodes;	        // List of nodes which use this interface.
```
Comment (PJungkamp) [inline include/villas/kernel/if.hpp:51]: My suggestion was using spaces instead of tabs for indentation. My bad.

Comment (stv0g) [reply include/villas/kernel/if.hpp]: Fixed :)

**Review by fwege — APPROVED** [review]

**Review by n-eiling — APPROVED** [review]

**Review by windrad6 — APPROVED** [review]

**Pull request merged**